Gems/ROS2/Code/Source/RobotImporter/ROS2RobotImporterEditorSystemComponent.cpp

```diff
@@ -61,4 +75,92 @@ namespace ROS2
         options.toolbarIcon = ":/ROS2/ROS_import_icon.svg";
         AzToolsFramework::RegisterViewPane<RobotImporterWidget>("Robot Importer", "ROS2", options);
     }
+
+    bool ROS2RobotImporterEditorSystemComponent::GeneratePrefabFromFile(
+        const AZStd::string_view filePath, bool importAssetWithUrdf, bool useArticulation)
+    {
+        if (filePath.empty())
+        {
+            AZ_Warning("ROS2EditorSystemComponent", false, "Empty file path");
+            return false;
+        }
+        if (Utils::IsFileXacro(filePath))
+        {
+            AZ_Warning("ROS2EditorSystemComponent", false, "XACRO is not supported");
+            return false;
+        }
+
+        urdf::ModelInterfaceSharedPtr parsedUrdf = UrdfParser::ParseFromFile(filePath);
+        if (!parsedUrdf)
+        {
+            AZ_Warning("ROS2EditorSystemComponent", false, "URDF parsing failed");
+            const auto log = UrdfParser::GetUrdfParsingLog();
+            AZ_Warning("ROS2EditorSystemComponent", false, "%s", log.c_str());
+            return false;
+        }
+
+        auto collidersNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), false, true);
+        auto visualNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), true, false);
+        auto meshNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), true, true);
+        AZStd::shared_ptr<Utils::UrdfAssetMap> urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>();
+        if (importAssetWithUrdf)
+        {
+            urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>(
+                Utils::CopyAssetForURDFAndCreateAssetMap(meshNames, filePath, collidersNames, visualNames));
+        }
+        bool allAssetProcessed = false;
+        do
+        {
+            allAssetProcessed = true;
+            for (const auto& [name, asset] : *urdfAssetsMapping)
+            {
+                auto sourceAssetFullPath = asset.m_availableAssetInfo.m_sourceAssetGlobalPath;
+                if (sourceAssetFullPath.empty())
+                {
+                    AZ_Warning("ROS2EditorSystemComponent", false, "asset %s is has no sourceAssetFullPath", name.c_str());
+                    continue;
```

Comment:
Looking at this logic, if there are no `sourceAssetFullPath` set in any of the `urdfAssetsMapping` table, then `allAssetsProcessed` will be true and this loop will complete. Is that correct?

Reply:
@arturkamieniecki

Reply:
It is correct @spham-amzn . The warning is produced since it is an abnormal situation.